src/sleep/sleep.c

```diff
@@ -211,6 +211,11 @@ static int execute(char **modes, char **states) {
                         return log_error_errno(r, "Failed to write mode to /sys/power/disk: %m");;
         }
 
+        char* action_env = strjoina("SYSTEMD_SLEEP_ACTION=", action);
```

Comment:
formatting nit since you have the docs to update: ```*``` nex to the variable name, not type